src/Veldrid/TextureViewDescription.cs

```diff
@@ -102,14 +121,54 @@ public TextureViewDescription(Texture target, uint baseMipLevel, uint mipLevels,
         /// <param name="mipLevels">The number of mip levels visible in the view.</param>
         /// <param name="baseArrayLayer">The base array layer visible in the view.</param>
         /// <param name="arrayLayers">The number of array layers visible in the view.</param>
-        public TextureViewDescription(Texture target, PixelFormat format, uint baseMipLevel, uint mipLevels, uint baseArrayLayer, uint arrayLayers)
+        /// <param name="viewType">The view type to create or <see cref="TextureViewType.Undefined"/> to detect from texture.</param>
+        public TextureViewDescription(Texture target, PixelFormat format, uint baseMipLevel, uint mipLevels, uint baseArrayLayer, uint arrayLayers, TextureViewType viewType = TextureViewType.Undefined)
         {
             Target = target;
+            if (viewType == TextureViewType.Undefined)
+                ViewType = GetFromTexture(target, target.Usage);
+            else
+                ViewType = viewType;
             BaseMipLevel = baseMipLevel;
             MipLevels = mipLevels;
             BaseArrayLayer = baseArrayLayer;
             ArrayLayers = arrayLayers;
-            Format = target.Format;
+            Format = format;
+        }
+
+        private static TextureViewType GetFromTexture(Texture target, TextureUsage usage)
+        {
+            switch (target.Type)
+            {
+                case TextureType.Texture1D:
+                    if (target.ArrayLayers > 1)
+                    {
+                        return TextureViewType.View1DArray;
+                    }
+
+                    return TextureViewType.View1D;
+                case TextureType.Texture2D:
+                    if ((usage & TextureUsage.Cubemap) == TextureUsage.Cubemap)
+                    {
+                        if (target.ArrayLayers > 1)
```

Comment:
nit: ternary expression would be better here (and is used elsewhere in related sections), e.g.

```C#
return target.ArrayLayers > 1 ? TextureViewType.ViewCubeArray : TextureViewType.ViewCube;
```

Same for the next section (View2DArray vs. View2D)